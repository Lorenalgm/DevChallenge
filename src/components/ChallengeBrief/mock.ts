export const brief = 'Escolha seu personagem preferido da Marvel!'
