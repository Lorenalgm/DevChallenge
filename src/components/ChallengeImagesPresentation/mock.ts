export const challengeImages = [
  'https://i.imgur.com/IroCUou.png',
  'https://i.imgur.com/efLHRzt.png'
]
