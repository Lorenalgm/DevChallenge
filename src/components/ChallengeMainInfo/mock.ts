import { ChallengeMainInfoProps } from '.'

export const challengeMainInfo: ChallengeMainInfoProps = {
  name: 'Marvel Heroes',
  description: 'Escolha seu personagem preferido da Marvel!',
  level: 'advanced',
  techs: ['Free choice'],
  type: 'Mobile',
  initialProjectUrl: 'https://github.com/Lorenalgm/marvel-heroes'
}
