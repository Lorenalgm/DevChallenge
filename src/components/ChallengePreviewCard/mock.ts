import { ChallengePreviewProps } from '.'

export const challengePreview: ChallengePreviewProps = {
  id: '1',
  name: 'Marvel Heroes',
  background: 'https://i.imgur.com/7FhmLIn.png',
  description: 'Escolha seu personagem preferido da Marvel!',
  level: 'advanced',
  techs: ['Free choice'],
  type: 'Mobile'
}
