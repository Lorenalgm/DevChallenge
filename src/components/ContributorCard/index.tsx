import {
  Flex,
  Heading,
  Avatar,
  Text,
  Stack,
  Link as ChakraLink
} from '@chakra-ui/react'
import { FaGithubSquare, FaLinkedin } from 'react-icons/fa'

export type ContributorProps = {
  id: string
  avatar: string
  name: string
  position: string
  github?: string
  linkedin?: string
}

const ContributorCard = ({
  avatar,
  name,
  position,
  github,
  linkedin
}: Omit<ContributorProps, 'id'>) => (
  <Flex
    justifyContent="space-between"
    alignItems="center"
    maxWidth="300px"
    borderRadius="md"
    paddingX="4"
    paddingY="5"
    color="white"
    bg="gray.800"
  >
    <Avatar name={name} src={avatar} />

    <Flex flexDirection="column" marginX="4">
      <Heading as="h4" fontSize="md">
        {name}
      </Heading>

      <Text fontSize="smaller" color="primary.400">
        {position}
      </Text>
    </Flex>

    <Stack>
      {github && (
        <ChakraLink
          href={`https://github.com/${github}`}
          isExternal
          _hover={{ color: 'primary.400' }}
        >
          <FaGithubSquare
            size="20"
            aria-hidden="false"
            aria-label={`${name} github profile`}
          />
        </ChakraLink>
      )}

      {linkedin && (
        <ChakraLink
          href={`https://www.linkedin.com/${linkedin}`}
          isExternal
          _hover={{ color: 'primary.400' }}
        >
          <FaLinkedin
            size="20"
            aria-hidden="false"
            aria-label={`${name} linkedin profile`}
          />
        </ChakraLink>
      )}
    </Stack>
  </Flex>
)

export default ContributorCard
