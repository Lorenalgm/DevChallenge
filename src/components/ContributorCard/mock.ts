import { ContributorProps } from '.'

export const contributor: ContributorProps = {
  id: '123456789',
  avatar: 'https://bit.ly/dan-abramov',
  name: 'Dan Abrahmov',
  position: 'Desenvolvedor',
  github: 'dan',
  linkedin: 'in/'
}
