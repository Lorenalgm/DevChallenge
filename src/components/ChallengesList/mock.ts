import { ChallengePreviewProps } from 'components/ChallengePreviewCard'

import { challengePreview } from 'components/ChallengePreviewCard/mock'

export const challenges: ChallengePreviewProps[] = [
  { ...challengePreview, id: '1' },
  { ...challengePreview, id: '2' },
  { ...challengePreview, id: '3' },
  { ...challengePreview, id: '4' },
  { ...challengePreview, id: '5' },
  { ...challengePreview, id: '6' }
]
