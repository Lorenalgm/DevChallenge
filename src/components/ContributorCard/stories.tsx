import { Story, Meta } from '@storybook/react/types-6-0'

import Wrapper from 'components/Wrapper'

import ContributorCard, { ContributorProps } from '.'

import { contributor } from './mock'

export default {
  title: 'ContributorCard',
  component: ContributorCard
} as Meta

export const Default: Story<ContributorProps> = (args) => (
  <Wrapper mt="10" alignItems="center">
    <ContributorCard {...args} />
  </Wrapper>
)

Default.args = { ...contributor }
