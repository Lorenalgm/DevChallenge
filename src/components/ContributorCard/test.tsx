import { render, screen } from 'utils/test-utils'

import ContributorCard from '.'

import { contributor } from './mock'

describe('<ContributorCard />', () => {
  it('should match snapshot', () => {
    const { container } = render(<ContributorCard {...contributor} />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render with user avatar', () => {
    render(<ContributorCard {...contributor} />)

    expect(
      screen.getByRole('img', { name: contributor.name })
    ).toBeInTheDocument()
  })

  it('should render with user name and position', () => {
    render(<ContributorCard {...contributor} />)

    expect(
      screen.getByRole('heading', { name: contributor.name })
    ).toBeInTheDocument()
    expect(screen.getByText(contributor.position)).toBeInTheDocument()
  })

  describe('when passed github username', () => {
    it('should render with github link', () => {
      render(<ContributorCard {...contributor} github="user" />)

      const githubLink = screen.getByRole('link', {
        name: `${contributor.name} github profile`
      })

      expect(githubLink).toBeInTheDocument()
      expect(githubLink).toHaveAttribute('href', `https://github.com/user`)
      expect(githubLink).toHaveAttribute('target', '_blank')
      expect(githubLink).toHaveAttribute('rel', 'noopener noreferrer')
    })
  })

  describe('when nt passed github username', () => {
    it('should not render with github link', () => {
      render(<ContributorCard {...contributor} github={undefined} />)

      expect(
        screen.queryByRole('link', {
          name: `${contributor.name} github profile`
        })
      ).not.toBeInTheDocument()
    })
  })

  describe('when passed linkedin username', () => {
    it('should render with linkedin link', () => {
      render(<ContributorCard {...contributor} linkedin="in/user" />)

      const linkedinLink = screen.getByRole('link', {
        name: `${contributor.name} linkedin profile`
      })

      expect(linkedinLink).toBeInTheDocument()
      expect(linkedinLink).toHaveAttribute(
        'href',
        `https://www.linkedin.com/in/user`
      )
      expect(linkedinLink).toHaveAttribute('target', '_blank')
      expect(linkedinLink).toHaveAttribute('rel', 'noopener noreferrer')
    })
  })

  describe('when nt passed linkedin username', () => {
    it('should not render with linkedin link', () => {
      render(<ContributorCard {...contributor} linkedin={undefined} />)

      expect(
        screen.queryByRole('link', {
          name: `${contributor.name} linkedin profile`
        })
      ).not.toBeInTheDocument()
    })
  })
})
