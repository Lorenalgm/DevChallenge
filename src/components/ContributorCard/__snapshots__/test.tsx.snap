// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContributorCard /> should match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  max-width: 300px;
  border-radius: var(--chakra-radii-md);
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  padding-top: var(--chakra-space-5);
  padding-bottom: var(--chakra-space-5);
  color: var(--chakra-colors-white);
  background: var(--chakra-colors-gray-800);
}

.emotion-1 {
  border-radius: var(--chakra-radii-full);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  text-align: center;
  text-transform: uppercase;
  font-weight: var(--chakra-fontWeights-medium);
  position: relative;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  background: #ab8353;
  color: var(--chakra-colors-gray-800);
  border-color: var(--chakra-colors-gray-800);
  vertical-align: top;
  width: var(--chakra-sizes-12);
  height: var(--chakra-sizes-12);
  font-size: calc(3rem / 2.5);
}

.emotion-2 {
  font-size: calc(3rem / 2.5);
  line-height: 3rem;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-margin-start: var(--chakra-space-4);
  margin-inline-start: var(--chakra-space-4);
  -webkit-margin-end: var(--chakra-space-4);
  margin-inline-end: var(--chakra-space-4);
}

.emotion-4 {
  font-family: var(--chakra-fonts-heading);
  font-weight: var(--chakra-fontWeights-bold);
  font-size: var(--chakra-fontSizes-md);
  line-height: 1.33;
}

@media screen and (min-width: 48em) {
  .emotion-4 {
    line-height: 1.2;
  }
}

.emotion-5 {
  font-size: smaller;
  color: var(--chakra-colors-primary-400);
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-6>*:not(style)~*:not(style) {
  margin-top: 0.5rem;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 0px;
  margin-inline-start: 0px;
}

.emotion-7 {
  -webkit-transition: all 0.15s ease-out;
  transition: all 0.15s ease-out;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 2px solid transparent;
  outline-offset: 2px;
  color: inherit;
}

.emotion-7:hover,
.emotion-7[data-hover] {
  color: var(--chakra-colors-primary-400);
}

.emotion-7:focus,
.emotion-7[data-focus] {
  box-shadow: var(--chakra-shadows-outline);
}

<div
  class="emotion-0"
>
  <span
    class="chakra-avatar emotion-1"
  >
    <div
      aria-label="Dan Abrahmov"
      class="chakra-avatar__initials emotion-2"
      role="img"
    >
      DA
    </div>
  </span>
  <div
    class="emotion-3"
  >
    <h4
      class="chakra-heading emotion-4"
    >
      Dan Abrahmov
    </h4>
    <p
      class="chakra-text emotion-5"
    >
      Desenvolvedor
    </p>
  </div>
  <div
    class="chakra-stack emotion-6"
  >
    <a
      class="chakra-link emotion-7"
      href="https://github.com/dan"
      rel="noopener noreferrer"
      target="_blank"
    >
      <svg
        aria-hidden="false"
        aria-label="Dan Abrahmov github profile"
        fill="currentColor"
        height="20"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zM277.3 415.7c-8.4 1.5-11.5-3.7-11.5-8 0-5.4.2-33 .2-55.3 0-15.6-5.2-25.5-11.3-30.7 37-4.1 76-9.2 76-73.1 0-18.2-6.5-27.3-17.1-39 1.7-4.3 7.4-22-1.7-45-13.9-4.3-45.7 17.9-45.7 17.9-13.2-3.7-27.5-5.6-41.6-5.6-14.1 0-28.4 1.9-41.6 5.6 0 0-31.8-22.2-45.7-17.9-9.1 22.9-3.5 40.6-1.7 45-10.6 11.7-15.6 20.8-15.6 39 0 63.6 37.3 69 74.3 73.1-4.8 4.3-9.1 11.7-10.6 22.3-9.5 4.3-33.8 11.7-48.3-13.9-9.1-15.8-25.5-17.1-25.5-17.1-16.2-.2-1.1 10.2-1.1 10.2 10.8 5 18.4 24.2 18.4 24.2 9.7 29.7 56.1 19.7 56.1 19.7 0 13.9.2 36.5.2 40.6 0 4.3-3 9.5-11.5 8-66-22.1-112.2-84.9-112.2-158.3 0-91.8 70.2-161.5 162-161.5S388 165.6 388 257.4c.1 73.4-44.7 136.3-110.7 158.3zm-98.1-61.1c-1.9.4-3.7-.4-3.9-1.7-.2-1.5 1.1-2.8 3-3.2 1.9-.2 3.7.6 3.9 1.9.3 1.3-1 2.6-3 3zm-9.5-.9c0 1.3-1.5 2.4-3.5 2.4-2.2.2-3.7-.9-3.7-2.4 0-1.3 1.5-2.4 3.5-2.4 1.9-.2 3.7.9 3.7 2.4zm-13.7-1.1c-.4 1.3-2.4 1.9-4.1 1.3-1.9-.4-3.2-1.9-2.8-3.2.4-1.3 2.4-1.9 4.1-1.5 2 .6 3.3 2.1 2.8 3.4zm-12.3-5.4c-.9 1.1-2.8.9-4.3-.6-1.5-1.3-1.9-3.2-.9-4.1.9-1.1 2.8-.9 4.3.6 1.3 1.3 1.8 3.3.9 4.1zm-9.1-9.1c-.9.6-2.6 0-3.7-1.5s-1.1-3.2 0-3.9c1.1-.9 2.8-.2 3.7 1.3 1.1 1.5 1.1 3.3 0 4.1zm-6.5-9.7c-.9.9-2.4.4-3.5-.6-1.1-1.3-1.3-2.8-.4-3.5.9-.9 2.4-.4 3.5.6 1.1 1.3 1.3 2.8.4 3.5zm-6.7-7.4c-.4.9-1.7 1.1-2.8.4-1.3-.6-1.9-1.7-1.5-2.6.4-.6 1.5-.9 2.8-.4 1.3.7 1.9 1.8 1.5 2.6z"
        />
      </svg>
    </a>
    <a
      class="chakra-link emotion-7"
      href="https://www.linkedin.com/in/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <svg
        aria-hidden="false"
        aria-label="Dan Abrahmov linkedin profile"
        fill="currentColor"
        height="20"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
        />
      </svg>
    </a>
  </div>
</div>
`;
